lib/compat/wordpress-5.8/block-editor.php

```diff
@@ -251,31 +256,32 @@ function gutenberg_get_default_block_editor_settings() {
  *
  * @since 10.5.0
  *
- * @param string $editor_name     The name of the editor (e.g. 'post-editor').
- * @param array  $custom_settings Optional custom settings to use with the editor type.
+ * @param array                   $custom_settings      Optional custom settings to use with the editor type.
+ * @param WP_Block_Editor_Context $block_editor_context The current block editor context.
  *
  * @return array The contextualized block editor settings.
  */
-function gutenberg_get_block_editor_settings( $editor_name, $custom_settings = array() ) {
+function gutenberg_get_block_editor_settings( $custom_settings, $block_editor_context ) {
```

Comment:
What is custom settings about? Should we just remove that argument and apply it outside the function

Reply:
I guess these are "additional" settings or something like that, maybe the order of the arguments should be reversed?

Reply:
I aligned with `gutenberg_block_editor_rest_api_preload` where we switched the order of params based on the feedback from @azaozz. We need to pass some `$custom_settings` to keep it backward compatible with how the post editor screen works. It exposes them to the filter together with all default settings.

Reply:
Oh, I need to adjust the description of the first param. It isn't optional anymore.